test/verify/run-tests

```diff
@@ -21,32 +82,124 @@ def check_valid(filename):
         return None
     return name.replace("-", "_")
 
+def build_command(filename, test, opts):
+    cmd = [filename]
+    if opts.sit:
+        cmd.append("-s")
+    if opts.trace:
+        cmd.append("-t")
+    if opts.verbosity:
+        cmd.append("-v")
+    if opts.thorough:
+        cmd.append("--thorough")
+    if not opts.fetch:
+        cmd.append("--nonet")
+    if opts.list:
+        cmd.append("-l")
+    cmd.append(test)
+    return cmd
 
 def run(opts):
-    # Now actually load the tests, any modules that start with "check-*"
-    loader = unittest.TestLoader()
-    suite = unittest.TestSuite()
+    # Build the list of tests we'll parallellize and the ones we'll run serially
+    test_loader = unittest.TestLoader()
+    parallel_tests = []
+    serial_tests = []
+    test_id = 1
+    result = 0
+    jobs = 1 if opts.list else opts.jobs
+
     for filename in glob.glob(os.path.join(os.path.dirname(__file__), "check-*")):
         name = check_valid(filename)
         if not name or not os.path.isfile(filename):
             continue
-        with open(filename, 'rb') as fp:
-            module = imp.load_module(name, fp, filename, ("", "rb", imp.PY_SOURCE))
-            suite.addTest(loader.loadTestsFromModule(module))
+        loader = importlib.machinery.SourceFileLoader(name, filename)
```

Comment:
just.... yikes!

Please execute the binary with `-l` and read the list that way, if you must...  What you're doing here is really bonkers....

Reply:
getting the list that way it takes about 8 seconds for the tests to be started vs negligible though :/

Reply:
The introspection needs to live somewhere anyway, to check for the `_testlib__non_destructive` and sorting into serial/parallel buckets, and do the pattern matching. From my POV, the extra loop to cover all check-* files is cleaner and probably even easier than having to encode and re-parse that from stdout.